SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -438,7 +438,7 @@ - (BOOL)disconnectSecondaryTransport {
     self.secondaryTransport = nil;
     self.secondaryProtocol = nil;
 
-    [self.backgroundTaskManager endBackgroundTask];
+    [self.backgroundTaskManager expiredTaskCleanupFinished];
```

Comment:
The task needs to be cleaned up even if the `if` above is true, right?

Reply:
Reverted to using `endBackgroundTask`. 

If the `if` statement is `true` then either the `disconnectSecondaryTransport` method was called twice or the `secondaryTransport` was never setup and thus a background task would not have been created as a background task is only created if an active TCP session is in progress.